testing/btest/language/type-expr.zeek

```diff
@@ -0,0 +1,19 @@
+# @TEST-DOC: Test valid use of type expressions in scripts
+# @TEST-EXEC: zeek -b %INPUT
+# @TEST-EXEC: TEST_DIFF_CANONIFIER= btest-diff .stdout
+
+global global_str = string;
+global my_global_string: global_str = "hey!!";
+
+event zeek_init()
+	{
+	local str = string;
+	local my_string: str = "hi there :)";
+	print my_string;
+	print my_global_string;
+
+	local integer = int;
+	local my_int: integer = 41;
+	my_int += 1;
+	print my_int;
+	}
```

Comment:
I presume it'll just work, but as one motivation was for bifs taking types, you could add another test for e.g. `from_json`:
```
zeek -e 'type str: string; print from_json("\"a\"", str)'
[v=a, valid=T]
zeek -e 'type countx: count; print from_json("1234", countx)'
[v=1234, valid=T]
```

Or `type_name(string)` could also be interesting :-)